nixos/modules/tasks/filesystems.nix

```diff
@@ -56,6 +56,16 @@ let
         type = types.listOf nonEmptyStr;
       };
 
+      depends = mkOption {
+        default = [ ];
+        example = [ "/persist" ];
+        type = types.listOf nonEmptyStr;
+        description = ''
+          List of paths that must be mounted before this one. Note that you do
+          not need to add the value of .device or .mountPoint to this list.
+        '';
```

Comment:
I suggest we explicitly require these options to be be `mountPoint`s since otherwise this suggests these will only be mounted once the specified paths exist.
See https://github.com/dotlambda/nixpkgs/commit/bdccb019df70fef86e250685c010b37068c7d162.

Reply:
I wrote it this way so that you can specify the paths you depend on rather than the mount points, which I believe is occasionally what you actually want. With this, you can write something like the following:
```nix
{
  fileSystems."/nix/.rw-store/work".device = "...";
}
```
and have it work without modifying `fileSystems."/nix/store"`. Granted, it's a contrived example, but I still believe it's better to do it this way.

Perhaps the description could be elaborated to make it clearer what the option does?
```
List of paths that should be mounted before this one. Note that you do
not need to add the value of .device or .mountPoint to this list.

If a path is added to this list, any other filesystem whose mount point is
a parent of the path will be mounted before this filesystem.
```

Reply:
How about you add an assumption that checks that all entries of `depends` are actually a suffix of some `mountPoint`.

Reply:
I'm not sure we actually want that though. This way you can just specify every path that is needed to mount a filesystem and have NixOS order your mounts automatically, regardless of whether any of them are actually needed.

One case where this might actually matter is if a mount's `depends` contains a path that may or may not need mounting depending on some configuration option.